android/src/main/kotlin/np/com/naxa/graphhooper_route/GraphhooperRoutePlugin.kt

```diff
@@ -49,24 +70,70 @@ class GraphhooperRoutePlugin : MethodCallHandler {
 
         var result: Result? = null
         var mapPath: String? = null
+        var downloadId: Long? = null
 
         constructor(result: Result, mapPath: String?) : this() {
             this.result = result
             this.mapPath = mapPath
         }
 
         private fun ifMapFileExists(): Boolean {
-            return getMapFolder().exists()
+            Log.d("IFMAPFILEEXISTS: ", File(Environment.getExternalStorageDirectory(), "/Download/nepal-gh/").exists().toString())
+            return File(Environment.getExternalStorageDirectory(), "/Download/nepal-gh").exists()
+        }
+
+        private fun downloadMapFromRemoteSource() {
+//            val baseUrl = ""
+//            val fileName = "nepal-gh.zip"
+//            val url = baseUrl
+//
+//            val request = DownloadManager.Request(Uri.parse(url))
+//            request.setAllowedNetworkTypes(DownloadManager.Request.NETWORK_MOBILE or DownloadManager.Request.NETWORK_WIFI)
+//            request.setTitle("Downloading $fileName")
+//            request.setDescription("Please wait")
+//            request.allowScanningByMediaScanner()
+//            request.setNotificationVisibility(DownloadManager.Request.VISIBILITY_VISIBLE_NOTIFY_COMPLETED)
+//            request.setDestinationInExternalPublicDir(Environment.DIRECTORY_DOWNLOADS,"nepal-gh")
+//
+//            val manager = ctx.applicationContext.getSystemService(Context.DOWNLOAD_SERVICE) as DownloadManager
+//            downloadId = manager.enqueue(request)
```

Comment:
Will implement these once we finalize where we will be hosting the file

Reply:
I have hosted it on firebase for now
https://firebasestorage.googleapis.com/v0/b/flutter-graphopper.appspot.com/o/nepal-gh.zip?alt=media&token=1026c9c3-b10f-43a2-b563-173c8ee556e6